src/ragas/metrics/_context_precision.py

```diff
@@ -90,6 +90,18 @@ class ContextPrecision(MetricWithLLM):
     evaluation_mode: EvaluationMode = EvaluationMode.qcg  # type: ignore
     context_precision_prompt: Prompt = field(default_factory=lambda: CONTEXT_PRECISION)
     max_retries: int = 1
+    _reproducibility: int = 1
+
+    @property
+    def reproducibility(self):
+        return self._reproducibility
```

Comment:
shouldn't this be part of `Metrics`?

Reply:
The question you have to ask is - does every `Metric` or `MetricWithLLM` or `MetricWithEmbedding` need this?

and you only have to ask the question if you have to do this 2+ times - which is this case you did. so what was the consideration here?

Reply:
I did think about this. For now, only 3/6 of the metrics can use this technique. I tried multiple ways to combine generative o/p too for improved reproducibility but I didn't get satisfactory results to raise a PR.  So, given that only 3/6 metrics have this wouldn't it be okay to keep it here now?